spring-kafka/src/main/java/com/baeldung/spring/kafka/streams/KafkaStreamsConfiguration.java

```diff
@@ -0,0 +1,153 @@
+package com.baeldung.spring.kafka.streams;
+
+import java.util.UUID;
+import java.util.function.BiConsumer;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.streams.KeyValue;
+import org.apache.kafka.streams.kstream.Grouped;
+import org.apache.kafka.streams.kstream.Joined;
+import org.apache.kafka.streams.kstream.KStream;
+import org.apache.kafka.streams.kstream.KTable;
+import org.apache.kafka.streams.kstream.Predicate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Bean;
+
+public class KafkaStreamsConfiguration {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(KafkaStreamsConfiguration.class);
+
+    //@Bean
+    public java.util.function.Consumer<KStream<String, String>> process() {
+        return inputStream ->
+            inputStream.foreach((key, value) -> {
+                LOGGER.info("Key: " + key + " Value: " + value);
+            });
+    }
+
+    //@Bean
+    public Function<KStream<String, String>, KStream<String, String>> process2() {
+        return inputStream -> inputStream
+            .map((key, value) -> {
+                LOGGER.info("Key: " + key + " Value: " + value);
+                return new KeyValue<>(key, value.toUpperCase());
+            });
+    }
+
+    @Bean
+    public BiFunction<KStream<Long, Long>, KTable<Long, Long>, KStream<Long, Long>> store() {
+        return (customerOrderStream, customerShopTable)
+            -> (customerOrderStream.leftJoin(customerShopTable, (count, shop)
+                -> new ShopOrders(shop == null ? 1L : shop, count),
```

Comment:
This double-nested use of lambda is pretty hard to read. I suggest extracting code to separate functions in order to make it readable

Reply:
Done